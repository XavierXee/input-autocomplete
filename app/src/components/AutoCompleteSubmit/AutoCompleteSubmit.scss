@import '../../_variables.scss';

.AutoCompleteSubmit {
  position: relative;
  margin: 0;
  color: $color_text;
  width: 25%;
  border: none;
  font-family: inherit;
  padding: $input-padding;
  font-size: $input-font-size;
  height: $input-height;
  background-color: $color_tertiary;
}