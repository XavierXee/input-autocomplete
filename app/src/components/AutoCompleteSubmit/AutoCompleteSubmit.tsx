import React from 'react';

import { AutoCompleteSubmitProps } from "../../interfaces/AutoCompleteSubmitProps";

import './AutoCompleteSubmit.scss';

class AutoCompleteSubmit extends React.Component<AutoCompleteSubmitProps> {
    constructor(props: any) {
        super(props);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    // TODO: type event
    handleSubmit(event: any): void {
        event.preventDefault();
        this.props.onSubmit();
    }

    render() {
        return (
            <button className="AutoCompleteSubmit" onClick={this.handleSubmit}>Find</button>
        );
    }
}

export default AutoCompleteSubmit;
