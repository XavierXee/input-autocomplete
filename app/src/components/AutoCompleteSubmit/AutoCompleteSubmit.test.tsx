import React from 'react';
import ReactDOM from 'react-dom';
import AutoCompleteSubmit from './AutoCompleteSubmit';
import { create } from "react-test-renderer";

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AutoCompleteSubmit onSubmit={() => {}}/>, div);
});

it('render correctly', () => {
    const component = create(<AutoCompleteSubmit onSubmit={() => {}}/>).toJSON();;
    expect(component).toMatchSnapshot();
});