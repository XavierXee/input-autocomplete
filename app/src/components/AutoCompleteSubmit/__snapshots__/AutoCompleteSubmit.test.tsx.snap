// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<button
  className="AutoCompleteSubmit"
  id="SearchButton"
  onClick={[Function]}
>
  Search
</button>
`;
